test.js

```diff
@@ -83,3 +84,8 @@ test('value iterator simple', t => {
   const r = fx([{val:1},{val:2}], '.[].val')
   t.deepEqual(JSON.parse(r), [1, 2])
 })
+
+test('reduce should replace keys with dash', t => {
+  const r = reduce({'release-it': 2}, 'this.release-it')
```

Comment:
```js
this.release-it
```
This is not valid JS. Use `this['release-it']` or `.['release-it']`.

Reply:
yes, but its from
```
const path = index.get(n)
    const subJson = reduce(json, 'this' + path)
```
and I rewrite it in reduce